ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/Server.java

```diff
@@ -324,4 +345,28 @@ private static HttpServerOptions createHttpServerOptions(final KsqlRestConfig ks
     return listeners;
   }
 
+  private static void configureTlsCertReload(
+      final KsqlRestConfig config,
+      final Set<ServerVerticle> serverVerticles
+  ) {
+    if (config.getBoolean(KsqlRestConfig.SSL_KEYSTORE_RELOAD_CONFIG)) {
+      final Path watchLocation;
+      if (!config.getString(KsqlRestConfig.SSL_KEYSTORE_WATCH_LOCATION_CONFIG).isEmpty()) {
+        watchLocation = Paths.get(
+            config.getString(KsqlRestConfig.SSL_KEYSTORE_WATCH_LOCATION_CONFIG));
+      } else {
+        watchLocation = Paths.get(config.getString(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG));
+      }
+
+      try {
+        FileWatcher.onFileChange(
+            watchLocation,
+            () -> serverVerticles.forEach(ServerVerticle::restartServer)
```

Comment:
I think this is more complex than it needs to be. Instead of individually starting and stopping each http server in each ServerVerticle I think it would be simpler just to add a restart() method to Server which simply stopped the Server and started it again, e.g.:

```
public void restart() {
  this.stop().thenCompose(this::start);
}
```

Reply:
This does not work. (I had this initially.) As explained in the PR description:
> It's necessary to recreate rather than simply restart the servers because Vert.x only loads TLS certs at the time that the KeyStoreHelper is created, which is only when servers are created.

Reply:
I'm referring to Server.start/stop not ServerVerticle.start/stop.

If you do that, it will take care of creating new Http Servers for you, no?

Reply:
Ah, I understand your suggestion now. We'll still need custom logic to ensure the HTTP servers are restarted with the same ports, in the event that port 0 is specified (as is the case in our tests). It's not clear to me that this is actually simpler.

Reply:
Here's a commit where I've made your suggested change: https://github.com/vcrfxia/ksql/commit/612b21595f7b87d2e6a389863d4fe1851cc239d6
The new test, `TlsTest#shouldReloadCert()` fails because the server ports changed during the restart but the client still tries to connect to the old port.

If we're OK with the ports changing on restart, we can update the test to recreate the client after the restart.

Reply:
Discussed offline. The only use case for setting port 0 at the moment is in our tests, so we're not going to worry about preserving ports across restarts. As such, I've moved the restart logic from ServerVerticle to Server as suggested, and updated the test to re-create clients in order to use the most up-to-date port.